cmd/eksctl/delete.go

```diff
@@ -71,34 +74,51 @@ func doDeleteCluster(cfg *eks.ClusterConfig, name string) error {
 
 	logger.Info("deleting EKS cluster %q", cfg.ClusterName)
 
-	debugError := func(err error) {
-		logger.Debug("continue despite error: %v", err)
+	handleError := func(err error) bool {
+		if err != nil {
+			logger.Debug("continue despite error: %v", err)
+			return true
+		}
+		return false
 	}
 
-	if err := ctl.DeleteControlPlane(); err != nil {
-		debugError(err)
+	// We can remove all 'DeprecatedDelete*' calls in 0.2.0
+
+	stackManager := ctl.NewStackManager()
+
+	if err := stackManager.WaitDeleteNodeGroup(); err != nil {
```

Comment:
There is a change in behaviour here. In previous versions when deleting a cluster it would kick of the stack deletion but not wait. Now we have the following behaviour:
- Wait for the deletion of the node group stack then
- Start the deletion of the cluster stack BUT don't wait

We should probably be consistent with the deletion of both stacks (i.e. not wait or wait). The behavaior before was to not wait. But i think waiting for both is good.

Reply:
We have to wait because nodegroup stack import values from the cluster stack, so cluster stack is actually protected until nodegroup stack is gone. Waiting for cluster to delete will take much longer, it takes basically at least as long as it take to create it (perhaps they have some kind of lazy deletion, like the EC2 instances which sit in terminated state for long time).

Reply:
We need to review deletions in general. Right the behaviour is a little weird, the output always says that your cluster will be delete even if the cluster doesn't exists. I can assure you this doesn't actually cause any distraction! :)

Reply:
Ah, that makes sense.